components/com_fabrik/models/form.php

```diff
@@ -2186,8 +2186,10 @@ public function validate()
 			foreach ($elementModels as $elementModel)
 			{
 				// If the user can't view or edit the element, then don't validate it. Otherwise user sees failed validation but no indication of what failed
-				if (!$elementModel->canUse() && !$elementModel->canView())
-				{
+				// if (!$elementModel->canUse() && !$elementModel->canView())
+				// Bauer note: This is wrong. For example, if !canUse, but canView, this would not be skipped because both are not false! 	
+				// It should be either... if (!$elementModel->canUse() || !$elementModel->canView()) - or...
```

Comment:
I wasn't sure what Hugh was getting at, but I would think so, no? - So long as either canUse or canView is false, the validation shouldn't be done at all, correct?

And I'm pretty sure that 
if ( !$elementModel->canUse() || !$elementModel->canView() )
...and...
if ( ! ( $elementModel->canUse() && $elementModel->canView() ) )
would do the same thing. So that's why I show both as replacements - take your pick.

But the original if condition...
if (!$elementModel->canUse() && !$elementModel->canView())
requires that BOTH conditions be true (well false actually because !==NOT)  before the 'continue' is met - when I'm pretty sure you want to continue (and not validate) if either condition is (!)met, no? 

Admittedly, it's one heck of a confusing and convoluted 'if' statement - NOT conditions in Boolean vars joined with ANDs or ORs usually are - but I think you get my point as to why it's not correct as stands.